modules/core/src/shaderlib/shadow/shadow.js

```diff
@@ -79,11 +80,9 @@ vec4 shadow_filterShadowColor(vec4 color) {
   }
   if (shadow_uUseShadowMap) {
     float shadowAlpha = 0.0;
-    for (int i = 0; i < max_lights; i++) {
-      if(i < int(shadow_uLightCount)) {
-        shadowAlpha += shadow_getShadowWeight(shadow_vPosition[i], shadow_uShadowMap[i]) * shadow_uColor.a / shadow_uLightCount;
-      }
-    }
+    shadowAlpha += shadow_getShadowWeight(shadow_vPosition[0], shadow_uShadowMap0);
+    shadowAlpha += shadow_getShadowWeight(shadow_vPosition[1], shadow_uShadowMap1);
```

Comment:
Do you need to check `shadow_uLightCount` here?

Reply:
no, if the count is less than one, the shadow_uUseShadowMap won't be true, this code snippet won't be executed.

Reply:
What if the count is 1? What is in `shadow_uShadowMap1`?